docs/agent_api.rst

```diff
@@ -3,4 +3,313 @@
+
+Also, users can access the underneath `Agent` implementation and fine-tune all of its parameters, options and behaviour in their final application.
+This is specially useful when dealing with a :ref:`custom_transport` implementation.
+
+Therefore, the API section is organized as follows:
+
+* :ref:`agent_instance_class_api`
+* :ref:`agent_class_api`
+
+.. _agent_instance_class_api:
+
+eprosima::uxr::AgentInstance
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The CLI manager, along with all the built-in agents available for the supported transports
+(UDP, TCP, Serial) are encapsulated in the `eprosima::uxr::AgentInstance` class.
+
+------
+
+.. code-block:: cpp
+
+    AgentInstance& getInstance();
+
+Gets a reference to the singleton `AgentInstance` wrapper class,
+which allows to launch Micro XRCE-DDS Agent upon user-given parameters.
+
+------
+
+.. code-block:: cpp
+
+    bool create(int argc, char** argv);
+
+Creates an UDP/TCP/Serial `Micro XRCE-DDS Agent`, based on the given arguments.
+
+Returns `true` if the arguments were valid and an agent was successfully created, or false otherwise.
+
+:argc: Number of arguments provided by the user via the CLI.
+       This is usually inherited from the `main` loop.
+:argv: List of arguments to be parsed by the CLI engine.
+
+------
+
+.. code-block:: cpp
+
+    bool run();
+
+Starts the execution of the previously created agent, until finished via an user interrupt or a process error.
+
+------
+
+.. code-block:: cpp
+
+    template <typename ... Args>
+    void add_middleware_callback(const Middleware::Kind& middleware_kind, const middleware::CallbackKind& callback_kind, std::function<void (Args ...)>&& callback_function);
+
+Sets an user-defined callback function for a specific create/delete middleware entity operation.
+
+:middleware_kind: Enum defining all the supported middlewares (see :ref:`middleware_abstraction_layer`).
+:callback_kind: Enum holding all the possible create/delete operations:
+
+.. code-block:: cpp
+
+    enum class CallbackKind : uint8_t
+    {
+        CREATE_PARTICIPANT,
+        CREATE_DATAWRITER,
+        CREATE_DATAREADER,
+        CREATE_REQUESTER,
+        CREATE_REPLIER,
+        DELETE_PARTICIPANT,
+        DELETE_DATAWRITER,
+        DELETE_DATAREADER,
+        DELETE_REQUESTER,
+        DELETE_REPLIER
+    };
+
+:callback_function: Callback to be defined by the user. It must follow a certain signature, depending on the used middleware.
+
+------
+
+.. _agent_class_api:
+
+eprosima::uxr::Agent
+^^^^^^^^^^^^^^^^^^^^
+
+However, it is also possible for users to create and instantiate their own agent instances, for example, to implement a :ref:`custom_transport_agent`.
+Also, in some scenarios it could be useful to have all the necessary `ProxyClient` and their associated DDS entities created by the agent even before
+clients are started, so that client applications get rid of the process of creating the session and the DDS entities and can focus only on the communication.
```

Comment:
```suggestion
*Clients* are started, so that *Clients* applications can avoid the process of creating the session and the DDS entities, and can focus on the communication.
```